tests/__init__.py

```diff
@@ -79,6 +79,17 @@ def decorator(func):
     return decorator
 
 
+def skip_if_lt_39(func):
+    def decorator(*args, **kwargs):
+
+        if sys.version_info < (3, 9):
+            pass
+        else:
+            func(*args, **kwargs)
+
+    return decorator
```

Comment:
Ok after so more thought, here's what I'd propose. Let's concretely define the closure of requirements. We know we need:

1. from importlib.resources  import files
2. from zipfile import Path

With both of those, are there any other concrete requirements? I may be missing one but let's define something like this in the project `__init__`.

```python
try:
    # These imports are generally only support in Python 3.9+
    from importlib.resources  import files
    from zipfile import Path
    HAS_ZIP_SUPPORT = True
except ImportError:
    HAS_ZIP_SUPPORT = False
```

Now we'll use this elsewhere to create a consistent restriction so we're not stuck with multiple disparate conditionals. I'll write them inline as separate comments.

Starting here, let's do something like:
```python
def requires_zip_support():
    reason = "Test requires zip dependencies available in Python 3.9+"
    def decorator(func):
        return unittest.skipIf(not HAS_ZIP_SUPPORT, reason)(func)
    return 
```

Reply:
The reason why I created this custom decorator was because I applied it to a unit test helper function, not a test itself. IIRC it wasn't working correctly, but maybe I'm misremembering. I'll refactor based on your suggestions.

Reply:
fwiw `importlib.resources` is only native as of 3.9 while `zipfile.Path` is as of 3.8 so that import line will never be reached when we expect the import error to be raised. should it still be included?